### PR TITLE
Add home image carousel component

### LINKED ISSUE #8: Homepage page Image Carousel

### PR DESCRIPTION
**Pull request by scuruchima1**
Screen Recording 2024-11-04 at 9 28 32 PM-2

Added an image carousel that fades in and out of given images. The small circles underneath the carousel can be used as buttons to switch to that image. Horizontal and vertical images stay within the same height as well.

### CODE REVIEW

**Comment by scuruchima1** [comment]
the rebasing lowkey messed up but its all good for merging

**Review by boldenj2 — CHANGES_REQUESTED** [review]
Looks good! I just want you to add a couple things. First, can you make the images in the carousel have the same height and width. Second, you also add some side buttons to scroll back and forth. Lastly, when you make prs can you add some more detail than just a video.

**Comment by scuruchima1** [comment]
Screenshot 2024-11-10 at 5 42 20 PM

This is the new look with small arrows on the bottom that allow for scrolling through the images. For images to have the same height I'm not sure if cropping images to a certain size is the way to go here, or if we should just get rid of vertical images completely in the carousel. 

I have also changed the component so it can be better reused. The component now takes in a list of image file names instead of having the file names hard coded within the component.

File: src/components/PhotoCarousal.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  const handleBackClick = () => {
+    setFade(false); // Trigger fade-out
+    setTimeout(() => {
+      setCurrentIndex((prevIndex) => (prevIndex - 1 + imageSrcs.length) % imageSrcs.length);
+      setFade(true); // Trigger fade-in after image change
+    }, 500);
+  };
+
+  const handleNextClick = () => {
+    setFade(false); // Trigger fade-out
```
Comment (boldenj2) [inline src/components/PhotoCarousal.js:47]: Reset auto-scroll timer.

File: src/components/PhotoCarousal.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    setTimeout(() => {
+      setCurrentIndex(index);
+      setFade(true); // Trigger fade-in after image change
+    }, 500); // Half-second fade-out before dot click update
+  };
+
+  const handleBackClick = () => {
+    setFade(false); // Trigger fade-out
+    setTimeout(() => {
```
Comment (boldenj2) [inline src/components/PhotoCarousal.js:39]: Reset auto-scroll timer.

File: src/components/PhotoCarousal.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        setFade(true); // Start fade-in after image change
+      }, 1000); // 1-second fade-out duration
+    }, 5000); // Image changes every 5 seconds
+
+    return () => clearInterval(interval);
+  }, [imageSrcs.length]);
+
+  const handleDotClick = (index) => {
+    setFade(false); // Trigger fade-out
+    setTimeout(() => {
```
Comment (boldenj2) [inline src/components/PhotoCarousal.js:30]: Reset auto-scroll timer.

**Review by boldenj2 — CHANGES_REQUESTED** [review]
Thank you for the changes, just one last thing. When I tested your code locally, I noticed the image carousel is a bit glitchy due to the auto-scroll timer not being reset. 

Example:
https://github.com/user-attachments/assets/d00b8771-f2f5-4916-a4fe-0535c08dca2c

**Comment by alognz** [comment]
lgtm but +1 the changes bolden said

File: src/components/PhotoCarousal.css [hunk 4]
```diff
@@ -0,0 +1,69 @@
+.carousel-container {
```
Comment (boldenj2) [inline src/components/PhotoCarousal.css]: Make sure this file is in the `styles` folder.

**Comment by scuruchima1** [comment]
The new commits now include a timer that resets on every back, forth, and image button click. The component style file has now been moved to the proper folder as well. :)

**Review by boldenj2 — APPROVED** [review]
Looks good!

**Pull request merged**